src/panels/lovelace/elements/hui-conditional-element.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Condition,
+  checkConditionsMet,
+  validateConditionalConfig,
+} from "../../lovelace/common/validate-condition";
+import { createConfiguredHuiElement } from "../../lovelace/cards/picture-elements/create-configured-hui-element";
+
+import { LovelaceElement, LovelaceElementConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+
+interface Config extends LovelaceElementConfig {
+  conditions: Condition[];
+  elements: LovelaceElementConfig[];
+}
+
+class HuiConditionalElement extends HTMLElement implements LovelaceElement {
+  public _hass?: HomeAssistant;
+  private _config?: Config;
+  private _elements: LovelaceElement[] = [];
+
+  public setConfig(config: Config): void {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.elements ||
+      !Array.isArray(config.elements) ||
+      !validateConditionalConfig(config.conditions)
+    ) {
+      throw new Error("Error in card configuration.");
+    }
+
+    this._config = config;
+
+    this._config.elements.map((elementConfig: LovelaceElementConfig) => {
+      this._elements.push(
```

Comment:
Why is this an array? You are only ever creating a single element.

Reply:
Ah see it now. that's fine.

You are still not clearing `this._elements` and cleaning up old elements.

Reply:
We support multiple elements under the same set of conditions